tests/common/helpers/parallel.py

```diff
@@ -77,3 +77,29 @@ def parallel_run(target, args, kwargs, nodes, timeout=None):
     logger.info('Completed running processes for target "{}" in {} seconds'.format(target.__name__, str(delta_time)))
 
     return results
+
+
+def reset_ansible_local_tmp(target):
+    """Decorator for resetting ansible default local tmp dir for parallel multiprocessing.Process
+
+    Args:
+        target (function): The function to be decorated.
+    """
+
+    def wrapper(*args, **kwargs):
+
+        # Reset the ansible default local tmp directory for the current subprocess
+        # Otherwise, multiple processes could share a same ansible default tmp directory and there could be conflicts
+        from ansible import constants
+        import os, shutil, tempfile
+        prefix = 'ansible-local-{}'.format(os.getpid())
+        constants.DEFAULT_LOCAL_TMP = tempfile.mkdtemp(prefix=prefix)
+
+        target(*args, **kwargs)
```

Comment:
Should we wrap this in a `try` block?

Reply:
IMO, exceptions raised by calling target should not be caught here, it should be handled by target itself or by outer block. But it is a good idea to use `try...finally` to run target and cleanup temp folder. Updated.

Reply:
Yes, I just want to remind the cleanup if an exception is raised.